R/DownloadModule.R

```diff
@@ -322,11 +318,301 @@ reporter_download_inputs <- function(rmd_yaml_args, rmd_output, showrcode, sessi
 #' @noRd
 #' @keywords internal
 any_rcode_block <- function(reporter) {
-  any(
-    vapply(
-      reporter$get_blocks(),
-      function(e) inherits(e, "RcodeBlock"),
-      logical(1)
+  cards <- reporter$get_cards()
+  if (all(vapply(cards, inherits, logical(1), "ReportCard"))) {
+    any(
+      vapply(
+        reporter$get_blocks(),
+        function(e) inherits(e, "RcodeBlock"),
+        logical(1)
+      )
     )
+  } else {
+    FALSE
+  }
+}
+
+
+
+report_render <- function(blocks, yaml_header, global_knitr = getOption("teal.reporter.global_knitr"), ...) {
+  tmp_dir <- tempdir()
+  output_dir <- file.path(tmp_dir, sprintf("report_%s", gsub("[.]", "", format(Sys.time(), "%Y%m%d%H%M%OS4"))))
+  dir.create(path = output_dir)
+
+  args <- list(...)
+  input_path <- report_render_Rmd(blocks, yaml_header, global_knitr, output_dir)
+  args <- append(args, list(
+    input = input_path,
+    output_dir = output_dir,
+    output_format = "all",
+    quiet = TRUE
+  ))
+  args_nams <- unique(names(args))
+  args <- lapply(args_nams, function(x) args[[x]])
+  names(args) <- args_nams
+
+  add_eval_false(input_path)
+  do.call(rmarkdown::render, args)
+  clean_chunks(input_path)
+  output_dir
+}
+
+add_eval_false <- function(input_rmd) {
+  lines <- readLines(input_rmd)
+
+  # Identify which chunks should NOT be modified
+  valid_chunks <- grepl("^```\\{r", lines) &      # Line starts with an R chunk
+    !grepl("object", lines) &      # Does NOT contain "object"
+    !grepl("eval\\s*=\\s*TRUE", lines) # Does NOT contain eval=TRUE
+
+  # Apply gsub only to valid chunks
+  lines[valid_chunks] <- gsub(
+    "(^```\\{r[^}]*)(\\})",  # Match `{r ... }`
+    "\\1, eval=FALSE\\2",    # Append `eval=FALSE`
+    lines[valid_chunks]
   )
+
+  writeLines(lines, input_rmd)
+}
+
+clean_chunks <- function(input_rmd) {
+  lines <- readLines(input_rmd)
+
+  new_lines <- c()
+  skip_chunk <- FALSE  # Track if we are inside an 'object' chunk
+
+  for (line in lines) {
+    # Start removing 'object' chunks
+    if (stringr::str_detect(line, "^```\\{r object_")) {
+      skip_chunk <- TRUE
+    }
+
+    # Stop skipping when the chunk ends
+    if (stringr::str_detect(line, "^```$") && skip_chunk) {
+      skip_chunk <- FALSE
+      next  # Skip adding this line
+    }
+
+    if (!skip_chunk) {
+      # Remove eval=FALSE from any R chunk
+      line <- stringr::str_replace(line, ",?\\s*eval=FALSE", "")
+
+      # Clean up extra commas in chunk headers
+      line <- stringr::str_replace(line, "\\{r,\\s*,", "{r,")
+      line <- stringr::str_replace(line, "\\{r,", "{r")  # Remove leftover commas
+
+      new_lines <- c(new_lines, line)
+    }
+  }
+
+  writeLines(new_lines, input_rmd)
+}
+
+report_render_Rmd <- function(blocks, yaml_header, global_knitr = getOption("teal.reporter.global_knitr"), output_dir) {
```

Comment:
I think `report_to_rmd` is more precise. Instead of `clean_chunks` regenerate Rmd without outputs.

```suggestion
report_to_rmd <- function(blocks, yaml_header, global_knitr = getOption("teal.reporter.global_knitr"), output_dir, include_echo = FALSE) {
```

Reply:
>  Instead of clean_chunks regenerate Rmd without outputs.

What do you mean by that?

Reply:
>  include_echo = FALSE

Do you want to pass `echo=FALSE` as a parameter to all chunks?

Reply:
> > Instead of clean_chunks regenerate Rmd without outputs.

> What do you mean by that?

in `report_render` you can `report_render_Rmd` to create a Rmd with outputs (as code_chunks) and then remove these code chunks from Rmd file.

```r
  input_path <- report_render_Rmd(blocks, yaml_header, global_knitr, output_dir)
  ...
  do.call(rmarkdown::render, args)
  clean_chunks(input_path)
```

instead you can do:

```r
  input_path <- report_render_Rmd(blocks, yaml_header, global_knitr, output_dir, include_echo = TRUE)
  ...
  do.call(rmarkdown::render, args)
  report_render_Rmd(blocks, yaml_header, global_knitr, output_dir, include_echo = FALSE)
```

Reply:
Brilliant 🔥

Reply:
I tried that in here a2a103b474459f3511ecda382961dbab79051a8d